File: CSharp/Contests2026/ABC464/B.cs
using static System.Linq.Enumerable;

class B
{
	static int[] Read() => Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
	static (int, int) Read2() { var a = Read(); return (a[0], a[1]); }
	static void Main() => Console.WriteLine(Solve());
	static object Solve()
	{
		var (h, w) = Read2();
		var c = Array.ConvertAll(new bool[h], _ => Console.ReadLine());

		int[] rh = [.. Range(0, h)];
		int[] rw = [.. Range(0, w)];

		var iMin = rh.Where(i => c[i].Contains('#')).Min();
		var iMax = rh.Where(i => c[i].Contains('#')).Max();
		iMax++;

		var jMin = rw.Where(j => rh.Any(i => c[i][j] == '#')).Min();
		var jMax = rw.Where(j => rh.Any(i => c[i][j] == '#')).Max();
		jMax++;

		var r = c[iMin..iMax].Select(s => s[jMin..jMax]);
		return string.Join("\n", r);
	}
}

File: Python/Contests2026/ABC464/B.py
h, w = map(int, input().split())
c = [input() for _ in range(h)]

i_min, i_max = h, 0
j_min, j_max = w, 0

for i in range(h):
    for j in range(w):
        if c[i][j] == "#":
            i_min = min(i_min, i)
            i_max = max(i_max, i)
            j_min = min(j_min, j)
            j_max = max(j_max, j)

for s in c[i_min:i_max + 1]:
    print(s[j_min:j_max + 1])
